mixer/pkg/protobuf/yaml/pool.go

```diff
@@ -12,23 +12,162 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// Copyright (c) 2016 Aliaksandr Valialkin, VertaMedia, Refactor by Andy Pan. All rights reserved.
```

Comment:
Because there is the proto.Buffer in pool of mixer while the bytebufferpool has its own [bytebuffer](https://github.com/valyala/bytebufferpool/blob/master/bytebuffer.go), those two are totally different kind of buffers.

Reply:
We could just add bytebufferpool as a dependency and use its inside [bytebuffer](https://github.com/valyala/bytebufferpool/blob/master/bytebuffer.go) whose internal []byte type could be exported as `B` which should be passed to `proto.Buffer.New([]byte)`, but that would be wired cuz it introduces an extra buffer type into mixer pool, doesn't it? or you think that is better than the current solution?